File: Daily Challanges/2026-06-June/2026-06-04.js
function isValidSchema(obj) {
  const roles = new Set(["user", "creator", "moderator", "staff", "admin"]);

  if (typeof obj.username !== "string") {
    return false;
  }

  if (typeof obj.posts !== "number") {
    return false;
  }

  if (typeof obj.verified !== "boolean") {
    return false;
  }

  if (!roles.has(obj.role)) {
    return false;
  }

  if ("supporter" in obj && typeof obj.supporter !== "boolean") {
    return false;
  }

  return true;
}

console.log(
  isValidSchema({
    username: "vivian",
    posts: 1,
    verified: false,
    role: "user",
    supporter: true,
  }),
);

console.log(
  isValidSchema({
    username: "rudolph",
    posts: 15,
    verified: true,
    role: "creator",
  }),
);

console.log(
  isValidSchema({
    username: "julia",
    posts: 50,
    verified: true,
    role: "admin",
    supporter: "true",
  }),
);

File: Daily Challanges/2026-06-June/2026-06-04.py
def is_valid_schema(obj):
    roles = {"user", "creator", "moderator", "staff", "admin"}

    if not isinstance(obj.get("username"), str):
        return False

    if not isinstance(obj.get("posts"), (int, float)) or isinstance(
        obj.get("posts"), bool
    ):
        return False

    if not isinstance(obj.get("verified"), bool):
        return False

    if obj.get("role") not in roles:
        return False

    if "supporter" in obj and not isinstance(obj["supporter"], bool):
        return False

    return True


print(
    is_valid_schema(
        {
            "username": "vivian",
            "posts": 1,
            "verified": False,
            "role": "user",
            "supporter": True,
        }
    )
)

print(
    is_valid_schema(
        {"username": "rudolph", "posts": 15, "verified": True, "role": "creator"}
    )
)

print(
    is_valid_schema(
        {
            "username": "julia",
            "posts": 50,
            "verified": True,
            "role": "admin",
            "supporter": "true",
        }
    )
)
